samples/snippets/src/test/java/com/example/storage/bucket/PrintBucketAclFilterByUserTest.java

```diff
@@ -0,0 +1,49 @@
+package com.example.storage.bucket;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.storage.Acl;
+import com.google.cloud.storage.Acl.Entity;
+import com.google.cloud.storage.Acl.Role;
+import com.google.cloud.storage.Acl.User;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.storage.testing.RemoteStorageHelper;
+import com.google.cloud.testing.junit4.StdOutCaptureRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class PrintBucketAclFilterByUserTest {
```

Comment:
So I have some thoughts around testing in this package. Currently there are individual tests per sample. Although given there are only 2 other samples we are in a position to change this.

1) In https://github.com/googleapis/google-cloud-java tests for multiple samples are in a single file. We could do this here as well.

2) We could create a TestBase which we extend in each of our tests to eliminate some repeated setup, teardown etc code and keep doing separate tests for each sample.

3) Continue with individual tests without a common base since its not too much repeat.

I am slightly partial to creating a test base but am not sure if its worth it since we are only repeating a few lines. Let me know any thoughts on this.

Reply:
I'm generally against mega test classes, too much gets intertwined in the setup and tear down over time.

I think option 2 seems good in this scenario since all the test follow a similar pattern ("Do something with a bucket, check std out for some expected tokens")